{"links": {"self": {"href": "data/repositories/osrf/release-tools/pullrequests/209/comments/17017862.json"}, "html": {"href": "#!/osrf/release-tools/pull-requests/209/_/diff#comment-17017862"}}, "parent": {"id": 17003871, "links": {"self": {"href": "data/repositories/osrf/release-tools/pullrequests/209/comments/17003871.json"}, "html": {"href": "#!/osrf/release-tools/pull-requests/209/_/diff#comment-17003871"}}}, "deleted": false, "pullrequest": {"type": "pullrequest", "id": 209, "links": {"self": {"href": "data/repositories/osrf/release-tools/pullrequests/209.json"}, "html": {"href": "#!/osrf/release-tools/pull-requests/209"}}, "title": "Rerun failed tests in gazebo homebrew (#60, #65)"}, "content": {"raw": "Breaking news update: I just figured out what the problem has been. The gazebo tests using QTestFixture will create the xml result file immediately and write to it after all the tests have finished. If the test seg-faults or timeouts, however, the xml file will be empty. I believe this was causing the `NullPointerException`s in the flaky test handler plugin.\n\nThe good news is that my pull request prevents the build from failing with `NullPointerException`'s. Those tests are displayed with a `.[empty]` suffix in the list of test results:\n\n* http://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64_flaky/14/testReport/\n\nThis testing also showed that the `flaky_junit_merge.py` script doesn't handle empty xml files well, giving `lxml.etree.XMLSyntaxError: None` and [breaking the build](http://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64_flaky/13/). So we should update this script.\n\nAlso, we could try fixing the upstream `check_test_ran.py` scripts to make sure they don't just check if a file exists, but also if it is non-empty.", "markup": "markdown", "html": "<p>Breaking news update: I just figured out what the problem has been. The gazebo tests using QTestFixture will create the xml result file immediately and write to it after all the tests have finished. If the test seg-faults or timeouts, however, the xml file will be empty. I believe this was causing the <code>NullPointerException</code>s in the flaky test handler plugin.</p>\n<p>The good news is that my pull request prevents the build from failing with <code>NullPointerException</code>'s. Those tests are displayed with a <code>.[empty]</code> suffix in the list of test results:</p>\n<ul>\n<li><a href=\"http://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64_flaky/14/testReport/\" rel=\"nofollow\" class=\"ap-connect-link\">http://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64_flaky/14/testReport/</a></li>\n</ul>\n<p>This testing also showed that the <code>flaky_junit_merge.py</code> script doesn't handle empty xml files well, giving <code>lxml.etree.XMLSyntaxError: None</code> and <a data-is-external-link=\"true\" href=\"http://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64_flaky/13/\" rel=\"nofollow\">breaking the build</a>. So we should update this script.</p>\n<p>Also, we could try fixing the upstream <code>check_test_ran.py</code> scripts to make sure they don't just check if a file exists, but also if it is non-empty.</p>", "type": "rendered"}, "created_on": "2016-04-10T20:36:11.083957+00:00", "user": {"display_name": "Steve Peters", "uuid": "{2ccfed09-18b8-4921-8d58-15ef01092802}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D"}, "html": {"href": "https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/1fb4816dad9e68337d3096f750951f6cd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsSP-1.png"}}, "nickname": "Steven Peters", "type": "user", "account_id": "557058:5de38267-b118-494c-aa76-4fab35448816"}, "updated_on": "2016-04-10T20:36:11.086545+00:00", "type": "pullrequest_comment", "id": 17017862}